builtin/providers/azurerm/resource_arm_loadbalancer_test.go

```diff
@@ -0,0 +1,193 @@
+	cases := []struct {
+		Value    string
+		ErrCount int
+	}{
+		{
+			Value:    "Random",
+			ErrCount: 1,
+		},
+		{
+			Value:    "Static",
+			ErrCount: 0,
+		},
+		{
+			Value:    "Dynamic",
+			ErrCount: 0,
+		},
+		{
+			Value:    "STATIC",
+			ErrCount: 0,
+		},
+		{
+			Value:    "static",
+			ErrCount: 0,
+		},
+	}
+
+	for _, tc := range cases {
+		_, errors := validateLoadbalancerPrivateIpAddressAllocation(tc.Value, "azurerm_lb")
+
+		if len(errors) != tc.ErrCount {
+			t.Fatalf("Expected the Azure RM Loadbalancer private_ip_address_allocation to trigger a validation error")
+		}
+	}
+}
+
+func TestAccAzureRMLoadbalancer_basic(t *testing.T) {
+	ri := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMLoadbalancerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMLoadbalancer_basic(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLoadbalancerExists("azurerm_lb.test"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMLoadbalancer_tags(t *testing.T) {
+	ri := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMLoadbalancerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMLoadbalancer_basic(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLoadbalancerExists("azurerm_lb.test"),
+					resource.TestCheckResourceAttr(
+						"azurerm_lb.test", "tags.%", "2"),
+					resource.TestCheckResourceAttr(
+						"azurerm_lb.test", "tags.Environment", "production"),
+					resource.TestCheckResourceAttr(
+						"azurerm_lb.test", "tags.Purpose", "AcceptanceTests"),
+				),
+			},
+			{
+				Config: testAccAzureRMLoadbalancer_updatedTags(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLoadbalancerExists("azurerm_lb.test"),
+					resource.TestCheckResourceAttr(
+						"azurerm_lb.test", "tags.%", "1"),
+					resource.TestCheckResourceAttr(
+						"azurerm_lb.test", "tags.Purpose", "AcceptanceTests"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMLoadbalancerExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		loadbalancerName := rs.Primary.Attributes["name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for loadbalancer: %s", loadbalancerName)
+		}
+
+		conn := testAccProvider.Meta().(*ArmClient).loadBalancerClient
+
+		resp, err := conn.Get(resourceGroup, loadbalancerName, "")
+		if err != nil {
+			return fmt.Errorf("Bad: Get on loadBalancerClient: %s", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
```

Comment:
In my experience the ArmClients raise an `err` for 404 responses while also filling the `resp` object.
A 404 response will actually trigger at line 117 and these lines will never be executed. It's a common error I've seen in other AzureRM acceptance tests.